### PR TITLE
Better flags, output and examples for hns-set

### PR DESCRIPTION
**Pull request by adrianludwin**
The main change here is to add --forbidCascadingDelete, as well as some
short flags (-p for --parent, -r for --root, -a for
--allowCascadingDelete and -f for --forbidCascadingDelete). It also adds
an "examples" section to the documentation and prints out fewer messages
if allowCD is unchanged.

Tested: manual testing for all the modified variants of 'set'; verified
that new example documentation looks good.

### CODE REVIEW

File: incubator/hnc/pkg/kubectl/set.go [hunk 1]
```diff
@@ -130,8 +166,9 @@ func normalizeStringArray(in []string) []string {
 }
 
 func newSetCmd() *cobra.Command {
-	setCmd.Flags().Bool("root", false, "Turns namespace into root namespace")
-	setCmd.Flags().String("parent", "", "Parent namespace")
-	setCmd.Flags().Bool("allowCascadingDelete", false, "Allows cascading deletion of its self-serve subnamespaces.")
+	setCmd.Flags().BoolP("root", "r", false, "Turns namespace into root namespace")
```
Comment (yiqigao217) [inline incubator/hnc/pkg/kubectl/set.go:169]: Nit not sure if it would be clearer to mention "removes current parent" for "root" flag.

Comment (adrianludwin) [reply incubator/hnc/pkg/kubectl/set.go]: Updated the descriptions, ptal!

**Pull request merged**